src/blocking/spi.rs

```diff
@@ -104,3 +104,28 @@ pub mod write_iter {
         }
     }
 }
+
+/// Operation for transactional SPI trait
+///
+/// This allows composition of SPI operations into a single bus transaction
+#[cfg(feature = "unproven")]
+#[derive(Debug, PartialEq)]
+pub enum Operation<'a, W: 'static> {
```

Comment:
is the static bound here okay? I can't think of a case it's _not_, but, could use `'a` instead if it was preferred

Reply:
Why use `'static` here?

This should ideally use a `Word` bound, but `Word` doesn't currently exist which makes it effectively impossible to portably use non-`u8` word sizes since there are no universally agreed-upon types for that.

Reply:
> Why use 'static here?

i was trying to be clear that it's a different bound on the type `W` vs. the references to the data in `Operation`, `'a` is also _correct_, but i couldn't think of a case where the `W` type wouldn't be static?

this is why the question ^_^

Reply:
the alternative also foists the lifetime bound to the trait, which is okay, but it's also kind of nice to have it in the function declaration (`fn exec<'a, O: AsMut<[Operation<'a, W>]>>(&mut self, operations: O) -> Result<(), Self::Error>`)

Reply:
In my limited experience, lifetime bounds are less problematic when attached to functions rather than traits as their scope is smaller but we probably will not add further functions to the trait which would need a different lifetime binding so it probably does not make much of a difference.